src/frontend/context/TidslinjeContext.ts

```diff
@@ -99,17 +100,45 @@ const [TidslinjeProvider, useTidslinje] = createUseContext(() => {
-                      return personMedAndelerTilkjentYtelse.ytelsePerioder.map(
-                          (ytelsePeriode: IYtelsePeriode, index: number) => ({
-                              fom: kalenderDatoTilDate(
+                      return personMedAndelerTilkjentYtelse.ytelsePerioder.reduce(
+                          (acc: Periode[], ytelsePeriode: IYtelsePeriode) => {
+                              const fom = kalenderDatoTilDate(
                                   hentFørsteDagIYearMonth(ytelsePeriode.stønadFom)
-                              ),
-                              tom: kalenderDatoTilDate(
-                                  hentSisteDagIYearMonth(ytelsePeriode.stønadTom)
-                              ),
-                              id: `${personMedAndelerTilkjentYtelse.personIdent}_${index}`,
-                              status: ytelsePeriode.beløp > 0 ? 'suksess' : 'feil',
-                          })
+                              );
+                              const periode: Periode = {
+                                  fom,
+                                  tom: kalenderDatoTilDate(
+                                      hentSisteDagIYearMonth(ytelsePeriode.stønadTom)
+                                  ),
+                                  id: `${
+                                      personMedAndelerTilkjentYtelse.personIdent
+                                  }_${fom.getMonth()}_${fom.getDay()}`,
+                                  status: ytelsePeriode.beløp > 0 ? 'suksess' : 'feil',
+                              };
+
+                              if (ytelsePeriode.ytelseType === YtelseType.UTVIDET_BARNETRYGD) {
+                                  const småbarnstilleggAndeler =
+                                      personMedAndelerTilkjentYtelse.ytelsePerioder.filter(
+                                          ytelsePeriodeFilter =>
+                                              ytelsePeriodeFilter.ytelseType ===
+                                              YtelseType.SMÅBARNSTILLEGG
+                                      );
+
+                                  return [
+                                      ...acc,
+                                      ...splittUtvidetVedEndringerPåSmåbarnstillegg(
+                                          periode,
+                                          ytelsePeriode,
+                                          småbarnstilleggAndeler
+                                      ),
+                                  ];
+                              } else if (ytelsePeriode.ytelseType !== YtelseType.SMÅBARNSTILLEGG) {
+                                  return [...acc, periode];
+                              } else {
+                                  return acc;
+                              }
+                          },
+                          []
                       );
                   }
               )
```

Comment:
For så vidt ikke en del av PR'en. Men litt rart at genererRader returnerer [[]] istede for bare [] hvis personerMedAndelerTilkjentYtelse er undefined. Altså linje 116

Reply:
på linje 145 (skulle skrevet det istede for 116, som er før endringene) så returneres det en liste med en tom liste inni. Returtypen er Periode[][], så da stussa jeg på hvorfor linje 145 returnerer [[]]. Altså hvorfor det returneres en tom liste av perioder inni lista. som skjer hvis personerMedAndelerTilkjentYtelse er undefined

Reply:
Ahh, seff!